### ISSUE TITLE
[iOS] Standalone jitsi-meet unable to connect to own server conference.

### ISSUE THREAD

**Issue by luismachado** [jitsi/jitsi-meet#2291]
I'm testing the standalone app on iOS (i'll probably integrate the SDKs on another app in the future) and I'm unable to connect to our own jitsi-meet server (@ https://live.pimentagroup.de/)

I always get the following error:
`console.error: [modules/xmpp/strophe.util.js], <n>: , Strophe: request id 1.1 error 0 happened
<unknown>
    polyfills-browser.js:262:45
error
    YellowBox.js:71:16
n
    lib-jitsi-meet.min.js:2:301140
log
    lib-jitsi-meet.min.js:2:702555
error
    lib-jitsi-meet.min.js:2:24699
_onRequestStateChange
    lib-jitsi-meet.min.js:2:54745
dispatchEvent
    event-target.js:172:43
setReadyState
    XMLHttpRequest.js:546:23
__didCompleteResponse
    XMLHttpRequest.js:387:25
emit
    EventEmitter.js:182:12
__callFunction
    MessageQueue.js:302:47
<unknown>
    MessageQueue.js:116:26
__guard
    MessageQueue.js:265:6
callFunctionReturnFlushedQueue
    MessageQueue.js:115:17
`

Furthermore, on the xcode console this also appears:
`2017-12-13 06:11:46.251 [info][tid:com.facebook.react.JavaScript] '[modules/xmpp/xmpp.js]', '<n>: ', '(TIME) Strophe connecting:\t', 1513145506250
2017-12-13 06:11:46.251295+0000 jitsi-meet[13760:5613004] '[modules/xmpp/xmpp.js]', '<n>: ', '(TIME) Strophe connecting:\t', 1513145506250
2017-12-13 06:11:46.262926+0000 jitsi-meet[13760:5612765] TIC TCP Conn Failed [370:0x1c0172d80]: 12:8 Err(-65554)
2017-12-13 06:11:46.263243+0000 jitsi-meet[13760:5612765] Task <72D65A56-C900-4051-99B3-C6DC864EACEE>.<3> HTTP load failed (error code: -1003 [12:8])
2017-12-13 06:11:46.263349+0000 jitsi-meet[13760:5611530] Task <72D65A56-C900-4051-99B3-C6DC864EACEE>.<3> finished with error - code: -1003
2017-12-13 06:11:46.311 [error][tid:com.facebook.react.JavaScript] '[modules/xmpp/strophe.util.js]', '<n>: ', 'Strophe: request id 1.1 error 0 happened'
2017-12-13 06:11:46.310799+0000 jitsi-meet[13760:5613004] '[modules/xmpp/strophe.util.js]', '<n>: ', 'Strophe: request id 1.1 error 0 happened'
2017-12-13 06:11:46.313 [warn][tid:com.facebook.react.JavaScript] '[modules/xmpp/strophe.util.js]', '<n>: ', 'Strophe: request errored, status: 0, number of errors: 1'
2017-12-13 06:11:46.313561+0000 jitsi-meet[13760:5613004] '[modules/xmpp/strophe.util.js]', '<n>: ', 'Strophe: request errored, status: 0, number of errors: 1'
2017-12-13 06:11:46.331573+0000 jitsi-meet[13760:5610767] TIC TCP Conn Failed [373:0x1c016c240]: 12:8 Err(-65554)
2017-12-13 06:11:46.331928+0000 jitsi-meet[13760:5613503] Task <F384DB82-6A9B-4B82-A624-AD3400DB0BA1>.<5> HTTP load failed (error code: -1003 [12:8])
2017-12-13 06:11:46.332099+0000 jitsi-meet[13760:5613503] Task <F384DB82-6A9B-4B82-A624-AD3400DB0BA1>.<5> finished with error - code: -1003
2017-12-13 06:11:46.333 [error][tid:com.facebook.react.JavaScript] '[modules/xmpp/strophe.util.js]', '<n>: ', 'Strophe: request id 2.2 error 0 happened'
2017-12-13 06:11:46.333064+0000 jitsi-meet[13760:5613004] '[modules/xmpp/strophe.util.js]', '<n>: ', 'Strophe: request id 2.2 error 0 happened'
2017-12-13 06:11:46.337 [warn][tid:com.facebook.react.JavaScript] '[modules/xmpp/strophe.util.js]', '<n>: ', 'Strophe: request errored, status: 0, number of errors: 2'
2017-12-13 06:11:46.337176+0000 jitsi-meet[13760:5613004] '[modules/xmpp/strophe.util.js]', '<n>: ', 'Strophe: request errored, status: 0, number of errors: 2'`

Any idea of what might be happening? Missing config on the server end? Thanks!

**Comment by saghul** [comment]
I think our mobile app won't like what you did there with `window.location.hostname` as the domain. Can you try to set it to the actual domain?

**Comment by luismachado** [comment]
Hello @saghul 

I didn't changed anything in the code. Just compiled it, ran it and pasted the URL of a conference room created on our server, which lead up to the aforementioned error (a jit.si room works thou).

**Comment by saghul** [comment]
Hey @luismachado!

You changed the config, which is an actual requirement. The default for `domain` is "jitsi-meet.example.com", but your has "window.location.hostname" there, which I think won't work on mobile. If you set it to "live.pimentagroup.de" (plus check in other settings on the config file) it should work.

**Comment by luismachado** [comment]
@saghul 

Thanks for the insight. So I tried to change the config.js file as you suggested (and also, for the sake of it, provided a URL on ViewController.m for view loadURL. The same error keeps popping in!

The strange thing is that I tried connecting to a different server (actually https://ctsivideo.mvoipctsi.com/ from another issue that popped in recently) and, without configuring anything (just downloaded the code, ran npm and pod install and launched the app) the with that URL it works!

Could it be anything related with the certificate?
